io/jvm-native/src/main/scala/fs2/io/net/SocketGroupPlatform.scala

```diff
@@ -105,16 +109,20 @@ private[net] trait SocketGroupCompanionPlatform { self: SocketGroup.type =>
       ): Stream[F, Socket[F]] = {
         def go: Stream[F, Socket[F]] = {
           def acceptChannel: F[AsynchronousSocketChannel] =
-            Async[F].async_[AsynchronousSocketChannel] { cb =>
-              sch.accept(
-                null,
-                new CompletionHandler[AsynchronousSocketChannel, Void] {
-                  def completed(ch: AsynchronousSocketChannel, attachment: Void): Unit =
-                    cb(Right(ch))
-                  def failed(rsn: Throwable, attachment: Void): Unit =
-                    cb(Left(rsn))
+            Async[F].async[AsynchronousSocketChannel] { cb =>
+              Async[F]
+                .delay {
+                  sch.accept(
+                    null,
+                    new CompletionHandler[AsynchronousSocketChannel, Void] {
+                      def completed(ch: AsynchronousSocketChannel, attachment: Void): Unit =
+                        cb(Right(ch))
+                      def failed(rsn: Throwable, attachment: Void): Unit =
+                        cb(Left(rsn))
+                    }
+                  )
                 }
-              )
+                .as(Some(Async[F].delay(sch.close())))
```

Comment:
There's potentially some weird life-cycle stuff going on here: AFAIK the only way to cancel an on-going `accept` is to close the server socket. As the signature is roughly `Resource[F, Stream[F, Socket]]`, where the server socket is tied to the resource, in theory you might expect to be able to cancel the inner stream and then start it again.

One way to fix this is to introduce a `Channel.synchronous` as an intermediary: that way, the stream can be canceled gracefully, without forcing to kill the server socket.

Thoughts?